src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TelemetryEventType.kt

```diff
@@ -1,18 +1,58 @@
 package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service
 
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+
 /**
- * An enumeration of the types of Goal events that can be sent to the Telemetry Service.
+ * An enumeration of the types of Goal events that can be sent to the Telemetry Service, where each item has:
+ *   * a text field `value` that is the name of the customEvent recorded in App Insights.
+ *   * a function `customDimensions` that returns the customDimensions that are recorded with the App Insights event.
  */
-enum class TelemetryEventType(val value: String) {
-  GOAL_CREATED("goal-created"),
-  GOAL_UPDATED("goal-updated"),
-  GOAL_STARTED("goal-started"),
-  GOAL_COMPLETED("goal-completed"),
-  GOAL_ARCHIVED("goal-archived"),
-  STEP_UPDATED("step-updated"),
-  STEP_NOT_STARTED("step-not-started"),
-  STEP_STARTED("step-started"),
-  STEP_COMPLETED("step-completed"),
-  STEP_REMOVED("step-removed"),
-  STEP_ADDED("step-added"),
+enum class TelemetryEventType(val value: String, val customDimensions: (goal: Goal) -> Map<String, String>) {
+  GOAL_CREATED(
+    "goal-created",
+    {
+      mapOf(
+        "status" to it.status.name,
+        "stepCount" to it.steps.size.toString(),
+        "reference" to it.reference.toString(),
+        "notesCharacterCount" to (it.notes?.length ?: 0).toString(),
+      )
+    },
+  ),
+
+  GOAL_UPDATED(
+    "goal-updated",
+    {
+      mapOf(
+        "reference" to it.reference.toString(),
+        "notesCharacterCount" to (it.notes?.length ?: 0).toString(),
+      )
+    },
+  ),
+
+  GOAL_STARTED("goal-started", { emptyMap() }),
```

Comment:
All these event types with an emptyMap of custom dimensions are event types that we are not currently sending (but are likely to be asked to; at which point we can implement the required custom dimensions for that particular event type 👍 )